### PR TITLE
Sarif v2

### PR DESCRIPTION
**Pull request by michaelcfanning**
@Evmaus-MS @pabrookes

### CODE REVIEW

**Comment by michaelcfanning** [comment]
Note that I will add v1 format compatibility in a follow on change. Also, we need a couple of SDK fixes (such as removing use of ruleMessageStringId'.

File: src/build.netcore.props [hunk 1]
```diff
@@ -5,6 +5,6 @@
   <PropertyGroup>
     <NetCoreVersion>netcoreapp2.0</NetCoreVersion>
     <NetStandardVersion>netstandard2.0</NetStandardVersion>
-	  <SarifSDKVersion>1.7.5</SarifSDKVersion>
+	  <SarifSDKVersion>2.0.0-csd.1</SarifSDKVersion>
```
Comment (Evmaus-MS) [inline src/build.netcore.props]: Nit: indentation (was already here)

**Comment by Evmaus-MS** [comment]
>             ILDiagnosticsAnalyzer roslynAnalyzer = ILDiagnosticsAnalyzer.Create(_globalRoslynAnalysisContext);

Do we want to just delete this?  This is the unused IL analysis code.  We can always tag the 1.5.0 release on master as the last release with it if we need to keep track of it.

---
Refers to: src/BinSkim.Driver/AnalyzeCommand.cs:110 in 1ddfd0b. [](commit_id = 1ddfd0bedaf2973ce76462a680aeff2950ae7404, deletion_comment = False)

**Comment by Evmaus-MS** [comment]
>     internal static class RoslynExtensionMethods

Same comment here about delete vs. keep.

---
Refers to: src/BinSkim.Driver/RoslynAnalyzer/RoslynExtensionMethods.cs:10 in 1ddfd0b. [](commit_id = 1ddfd0bedaf2973ce76462a680aeff2950ae7404, deletion_comment = False)

**Pull request merged**